pkg/devfile/adapters/common/command.go

```diff
@@ -134,17 +147,32 @@ func GetRunCommand(data data.DevfileData, devfileRunCmd string) (runCommand comm
 // ValidateAndGetPushDevfileCommands validates the build and the run command,
 // if provided through odo push or else checks the devfile for devBuild and devRun.
 // It returns the build and run commands if its validated successfully, error otherwise.
-func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
+func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileInitCmd, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
 	var emptyCommand common.DevfileCommand
-	isBuildCommandValid, isRunCommandValid := false, false
+	isInitCommandValid, isBuildCommandValid, isRunCommandValid := false, false, false
+
+	initCommand, initCmdErr := GetInitCommand(data, devfileInitCmd)
+
+	isInitCmdEmpty := reflect.DeepEqual(emptyCommand, initCommand)
+	if isInitCmdEmpty && initCmdErr == nil {
+		// If there was no init command specified through odo push and no default init command in the devfile, default validate to true since the init command is optional
+		isInitCommandValid = true
+		glog.V(3).Infof("No init command was provided")
```

Comment:
shouldn't be using `v(3)`

Reply:
Is there a guideline for using different log levels?

Reply:
I believe odo mostly uses `v(4)`

Reply:
Okay, thanks! The existing code used `v(3)`, so we just copied the standard there. I'll update them now though.